Jenkinsfile

```
@@ -1,31 +1,139 @@
+import org.apache.tools.ant.types.selectors.SelectorUtils
+import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+
 /*
 ** Variables.
 */
 properties([buildDiscarder(logRotator(numToKeepStr: '50'))])
 def serie = '21.04'
 def maintenanceBranch = "${serie}.x"
+env.REF_BRANCH = 'master'
 env.PROJECT='centreon-web'
 if (env.BRANCH_NAME.startsWith('release-')) {
   env.BUILD = 'RELEASE'
-} else if ((env.BRANCH_NAME == 'master') || (env.BRANCH_NAME == maintenanceBranch)) {
+} else if ((env.BRANCH_NAME == env.REF_BRANCH) || (env.BRANCH_NAME == maintenanceBranch)) {
   env.BUILD = 'REFERENCE'
 } else {
   env.BUILD = 'CI'
 }
+
+def buildBranch = env.BRANCH_NAME
+if (env.CHANGE_BRANCH) {
+  buildBranch = env.CHANGE_BRANCH
+}
+
+def backendFiles = [
+  'Jenkinsfile',
+  '**/*.php',
+  '**/*.js',
+  '**/*.sh',
+  'composer.*',
+  'symfony.lock',
+  '.env*',
+  'behat.yml',
+  'codingstyle.xml',
```

this file is generated from the analysis and added (or should be added) to the .gitignore
```suggestion
  'ruleset.xml',
```